### DISCUSSION TITLE
PYBIND11_NUMPY_DTYPE equivalent in nanobind

### DISCUSSION THREAD

**Discussion by zhqrbitee** [wjakob/nanobind#271] [Q&A] [answered]
Hi,
Is there any equivalent of `PYBIND11_NUMPY_DTYPE` in nanobind? 
We have some codes that takes advantage of pybind11 structured types: https://pybind11.readthedocs.io/en/stable/advanced/pycpp/numpy.html#structured-types, but I didn't find any equivalent method in nanobind docs.

**Comment by wjakob** [comment, marked as answer]
No, and such a feature also will not be added (see the docs)

**Reply by zhqrbitee** [reply]
Thanks for the quick reply!

**Comment by dariobaron** [comment]
> No, and such a feature also will not be added (see the docs)

I understand (and hugely appreciate) the philosophy behind the library design and the choice of DLPack.
Nevertheless, I wonder how extensive changes to the implementation would be to be able to deal with mixed types in nb::ndarray. In some cases, (among those I could mention the scientific computing, where very often happens to work with very large datasets of mixed types) it would be awesome to have this feature, and recurring to Pybind only for this aspect might be a pity.
Could you motivate the choice of non-including this specific feature?
